core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -309,20 +312,28 @@ class KafkaServer(
         socketServer = new SocketServer(config, metrics, time, credentialProvider, apiVersionManager)
 
         // Start alter partition manager based on the IBP version
-        alterIsrManager = if (config.interBrokerProtocolVersion.isAlterPartitionSupported) {
+        alterPartitionManager = if (config.interBrokerProtocolVersion.isAlterPartitionSupported) {
+          val alterPartitionChannelManager = BrokerToControllerChannelManager(
```

Comment:
nit: I think I favor creating the `BrokerToControllerChannelManager` in `AlterPartitionManager.apply`. The main thing is that it makes the ownership clearer. Currently `DefaultAlterPartitionManager` is responsible for starting and stopping the channel manager. We can revise `BrokerServer` to use the same `apply` method (I am not sure why we didn't do that).

Reply:
I guess we create `BrokerToControllerChannelManager` in `BrokerServer` because we want to consolidate these channel to use only one, https://github.com/apache/kafka/pull/10135#discussion_r577322703
but it's it's still early to consolidate channel now so I moved it to `AlterPartitionManager.apply` as you suggested.

Reply:
Consolidation is a little difficult because of head-of-line blocking (sigh). It might be reasonable to combine the alter partition and broker lifecycle channel managers though. Anyway, this is definitely a separate patch.